torch/optim/adagrad.py

```diff
@@ -77,14 +100,17 @@ def __setstate__(self, state):
             group.setdefault("foreach", None)
             group.setdefault("maximize", False)
             group.setdefault("differentiable", False)
+            fused = group.setdefault("fused", None)
 
         state_values = list(self.state.values())
         step_is_tensor = (len(state_values) != 0) and torch.is_tensor(
             state_values[0]["step"]
         )
         if not step_is_tensor:
             for s in state_values:
-                s["step"] = torch.tensor(float(s["step"]), dtype=_get_scalar_dtype())
+                s["step"] = torch.tensor(
+                    float(s["step"]), dtype=_get_scalar_dtype(is_fused=fused)
```

Comment:
this is used when the fused impl only handles fp32 for step--is this true in the cpp kernel?

Reply:
Yes, we are following CPU kernel for `SGD/Adam/AdamW` here. And CPU kernel is following CUDA impl.